python/cudf/cudf/tests/test_series.py

```diff
@@ -2224,3 +2224,14 @@ def __getitem__(self, key):
 def test_series_constructor_error_mixed_type():
     with pytest.raises(pa.ArrowTypeError):
         cudf.Series(["abc", np.nan, "123"], nan_as_null=False)
+
+
+def test_series_typecast_error():
+    actual = cudf.Series([1, 2, 3], dtype="datetime64[ns]")
+    with cudf.option_context("mode.pandas_compatible", True):
```

Comment:
Should we add tests that this does _not_ fail when `pandas_compatible` is `False`? (With the expectation that behavior could change in the future)

Reply:
Done 👍